src/Client.php

```diff
@@ -275,4 +275,151 @@ public function getAsset($assetId) {
     return (string) $response->getBody();
   }
 
+  /**
+   * Gets presigned url from AWS S3.
+   *
+   * @param array $file_data
+   *   The file data required by Webdam.
+   *
+   * @return mixed
+   *   Presigned url needed for next step + PID.
+   */
+  public function getPresignUrl(array $file_data) {
+    $this->checkAuth();
+
+    try {
```

Comment:
This should not be wrapped in a try/catch. We want to bubble the ClientException up to the caller.